error: use of undeclared identifier 'mean'

## Specifications
- OS: Mac 12.1
- SCS Version: 3.1.0
- Compiler: Apple clang version 13.0.0 (clang-1300.0.29.30)

## Description

Compilation fails with SCS 3.1.0 and `USE_LAPACK=0` with `error: use of undeclared identifier 'mean'`. Changing line 88 to `scs_float mean = 0.0;` fixes it.

## How to reproduce

```sh
make USE_LAPACK=0
```

## Output

```text
cc -g -Wall -Wwrite-strings -pedantic -funroll-loops -Wstrict-prototypes -I. -Iinclude -Ilinsys -O3 -fPIC  -DCTRLC=1  -DCOPYAMATRIX=1  -DGPU_TRANSPOSE_MAT=1  -DINDIRECT=1 -c src/scs.c -o src/scs_indir.o
cc -g -Wall -Wwrite-strings -pedantic -funroll-loops -Wstrict-prototypes -I. -Iinclude -Ilinsys -O3 -fPIC  -DCTRLC=1  -DCOPYAMATRIX=1  -DGPU_TRANSPOSE_MAT=1    -c -o src/util.o src/util.c
cc -g -Wall -Wwrite-strings -pedantic -funroll-loops -Wstrict-prototypes -I. -Iinclude -Ilinsys -O3 -fPIC  -DCTRLC=1  -DCOPYAMATRIX=1  -DGPU_TRANSPOSE_MAT=1    -c -o src/cones.o src/cones.c
cc -g -Wall -Wwrite-strings -pedantic -funroll-loops -Wstrict-prototypes -I. -Iinclude -Ilinsys -O3 -fPIC  -DCTRLC=1  -DCOPYAMATRIX=1  -DGPU_TRANSPOSE_MAT=1    -c -o src/aa.o src/aa.c
cc -g -Wall -Wwrite-strings -pedantic -funroll-loops -Wstrict-prototypes -I. -Iinclude -Ilinsys -O3 -fPIC  -DCTRLC=1  -DCOPYAMATRIX=1  -DGPU_TRANSPOSE_MAT=1    -c -o src/rw.o src/rw.c
cc -g -Wall -Wwrite-strings -pedantic -funroll-loops -Wstrict-prototypes -I. -Iinclude -Ilinsys -O3 -fPIC  -DCTRLC=1  -DCOPYAMATRIX=1  -DGPU_TRANSPOSE_MAT=1    -c -o src/linalg.o src/linalg.c
src/linalg.c:88:3: error: use of undeclared identifier 'mean'
  mean = 0.;
  ^
src/linalg.c:90:5: error: use of undeclared identifier 'mean'
    mean += x[i];
    ^
src/linalg.c:92:10: error: use of undeclared identifier 'mean'
  return mean / n;
         ^
3 errors generated.
make: *** [src/linalg.o] Error 1
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -9,13 +9,14 @@ jobs:
       fail-fast: false
       matrix:
         long: [0, 1]
+        lapack: [0, 1]
 
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: sudo apt-get install libopenblas-dev liblapack-dev
-      - run: make DLONG=${{ matrix.long }}
-      - run: make test DLONG=${{ matrix.long }}
+      - run: make DLONG=${{ matrix.long }} USE_LAPACK=${{ matrix.lapack }}
+      - run: make test DLONG=${{ matrix.long }} USE_LAPACK=${{ matrix.lapack }}
       - run: out/run_tests_direct    # test direct solver
       - run: out/run_tests_indirect  # test indirect solver
 
@@ -32,12 +33,13 @@ jobs:
       fail-fast: false
       matrix:
         long: [0, 1]
+        lapack: [0, 1]
 
     runs-on: macos-latest
     steps:
       - uses: actions/checkout@v2
       - run: brew install openblas lapack
-      - run: make DLONG=${{ matrix.long }}
-      - run: make test DLONG=${{ matrix.long }}
+      - run: make DLONG=${{ matrix.long }} USE_LAPACK=${{ matrix.lapack }}
+      - run: make test DLONG=${{ matrix.long }} USE_LAPACK=${{ matrix.lapack }}
       - run: out/run_tests_direct    # test direct solver
       - run: out/run_tests_indirect  # test indirect solver

--- a/src/linalg.c
+++ b/src/linalg.c
@@ -85,7 +85,7 @@ void SCS(add_scaled_array)(scs_float *a, const scs_float *b, scs_int n,
 
 scs_float SCS(mean)(const scs_float *x, scs_int n) {
   scs_int i;
-  mean = 0.;
+  scs_float mean = 0.;
   for (i = 0; i < n; ++i) {
     mean += x[i];
   }

--- a/test/run_tests.c
+++ b/test/run_tests.c
@@ -11,19 +11,28 @@
 #include "problems/hs21_tiny_qp_rw.h"
 #include "problems/infeasible_tiny_qp.h"
 #include "problems/qafiro_tiny_qp.h"
-#include "problems/rob_gauss_cov_est.h"
 #include "problems/small_lp.h"
 #include "problems/small_qp.h"
 #include "problems/test_validation.h"
 #include "problems/unbounded_tiny_qp.h"
 
-#ifndef DLONG /* TODO: this reads a file written with 32bit ints */
+#define _skip(problem)                                                         \
+  char *problem(void) {                                                        \
+    scs_printf("skipped\n");                                                   \
+    return 0;                                                                  \
+  }
+
+#ifdef USE_LAPACK /* solve SDPs, requires blas / lapack */
+#include "problems/rob_gauss_cov_est.h"
+#else
+_skip(rob_gauss_cov_est)
+#endif
+
+/* TODO: this reads a file written with 32bit ints */
+#if defined(USE_LAPACK) && !defined(DLONG)
 #include "problems/random_prob.h"
 #else
-char * random_prob(void) {
-  scs_printf("skipped\n"); 
-  return 0;
-}
+    _skip(random_prob)
 #endif
 
 int tests_run = 0;

